### ISSUE TITLE
RP silently blocks remote content with Protonmail

### ISSUE THREAD

**Issue by Watilin** [RequestPolicyContinued/requestpolicy#850]
__Browser and Add-ons__

- Browser: Firefox 55.0.2
- RequestPolicy version: 1.0.beta13.0
- No other add-ons

RequestPolicy settings:
- Default policy: "allow"

__Steps to reproduce__

1. Create a new browser profile
2. Install above add-ons
3. Configure the add-ons as described above
4. If not already done, register an account on protonmail.com – it’s free. Protonmail is an end-to-end encrypted email service and webclient.
5. Log in to protonmail.com
6. Select an email containing one or more images. If you don’t have one, send yourself an email containing, for example, https://github.com/favicon.ico
7. Protonmail blocks remote content by default, but exposes a button to unblock it. Click this button.

__What happens?__

No image appears in the email. The Request Policy icon does not turn red, and the menu does not mention anything blocked.

__What should happen?__

The menu should tell what content Request Policy has blocked, and offer a way to unblock it.

Another experiment makes it possible to be certain that Request Policy is what blocks the image once you click on Protonmail’s unblock button:

8. Select another email to hide your test email away
9. Completely disable Request Policy blocking (icon must turn orange)
10. Go back your test email, the unblock button should be back
11. Click the unblock button

Now you should see the images.

Due to the encrypted nature of Protonmail, I’m not sure how Request Policy determines what is remote content; yet it manages to block it somehow.

The workaround I’m using is to allow *any* request from Protonmail.com, but this lacks fine-grain tuning.

**Comment by myrdd** [comment]
Hi @Watilin, thank you for your detailed report. It would be great if you could post a screenshot of RPC's request log after reproducing step 7. Thank you. :)

**Comment by myrdd** [comment]
the request log is not documented yet, but you can see here how it works: https://github.com/RequestPolicyContinued/requestpolicy/issues/853

**Comment by Watilin** [comment]
Hi! Sorry for the waiting.

I’m bringing a full new report using Pale Moon, with a capture showing the request log as asked— in hope that this will shed some light on the situation.

__Browser and Add-ons__

- Browser: Pale Moon 27.6.1 (32bits)
- RequestPolicy version: 1.0.beta13.2
- No other add-ons

RequestPolicy settings:
- Default policy: "block" (allow same domain)

**Please note: steps to reproduce and results are exactly the same as before, I’m just copy-pasting for convenience.**

---

__Steps to reproduce__

1. Create a new browser profile
2. Install above add-ons
3. Configure the add-ons as described above
4. If not already done, register an account on protonmail.com – it’s free. Protonmail is an end-to-end encrypted email service and webclient.
5. Log in to protonmail.com
6. Select an email containing one or more images. If you don’t have one, send yourself an email containing, for example, https://github.com/favicon.ico
7. Protonmail blocks remote content by default, but exposes a button to unblock it. Click this button.

__What happens?__

No image appears in the email. The Request Policy icon does not turn red, and the menu does not mention anything blocked.

__What should happen?__

The menu should tell what content Request Policy has blocked, and offer a way to unblock it.

---

The blocked image *does* appear in the request log. There be the screen capture :)
![capture](https://i.imgur.com/4JsWror.png)

**Comment by myrdd** [comment]
Confirmed, see also #867.
The problem is: The address bar shows a different url than the "origin" field in the request log.
You can work around this issue by reloading the page using `Ctrl+R`.
Please wait until the WebExtension porting is finished, maybe this issue will be fixed as a side effect of the WE port.
